engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -23,7 +23,7 @@ import scala.util.{Try, Failure, Success}
 object ValidateActor {
   private val tag = "ValidateActor"
 
-  def props(wdlSource: WdlSource, wdlJson: Option[WdlJson], workflowOptions: Option[WdlJson]): Props = {
+  def props(wdlSource: WdlSource, wdlJson: Option[WdlJson] = None, workflowOptions: Option[WdlJson] = None): Props = {
```

Comment:
@Horneth What you're describing in your 2nd paragraph is really more about how to manage transactional behavior, and you're right that a single actor managing other things isn't the right place for it. Take a look (if you haven't already) at the saga pattern, which I think you'll find is not dissimilar to your DB POC.

One way of implementing it in akka would be to have the actor which wants a multi-stage transaction to happen spin up an actor for that specific transaction. When it is done it kills itself and the main actor (e.g. the WorkflowActor) sees that the child has died and knows that transaction is complete.